core/input/input.h

```diff
@@ -154,7 +154,7 @@ class Input : public Object {
 		StringName name;
 		StringName uid;
 		bool connected;
-		bool last_buttons[JOY_BUTTON_MAX + 19]; //apparently SDL specifies 35 possible buttons on android
+		bool last_buttons[JOY_BUTTON_MAX];
 		float last_axis[JOY_AXIS_MAX];
```

Comment:
Looking at it a bit more, there is definitely a need to separate out the SDL mappings MAX values from the general joy MAX values.